### PR TITLE
Feature/docker image

### PR DESCRIPTION
**Pull request by jbygdell**
Changes made:
* switch to alpine based image
* Add metadata and build info to docker image
* Add build hooks used by docker hub

Related to: EGA-archive/LocalEGA#23

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,13 @@
-FROM rabbitmq:3.6.14-management
+FROM rabbitmq:3.6-management-alpine
```
Comment (viklund) [inline Dockerfile]: Just a small question. I can't find the v3.6 images listed as a "supported" tag on docker hub. Is this something we should be concerned with? https://hub.docker.com/_/rabbitmq?tab=description&page=7

Comment (jbygdell) [reply Dockerfile]: Tags | 3.6.16-management-alpine 3.6-management-alpine

**Comment by silverdaz** [comment]
I still don't think the build hook is necessary.
We can get around with the docker hub features.
Remove the hook.

**Comment by jbygdell** [comment]
> I still don't think the build hook is necessary.
> We can get around with the docker hub features.
> Remove the hook.

Without the hook the `BUILD_DATE` and `SOURCE_COMMIT` where not populated when I tested automated builds on Docker Hub.

**Comment by silverdaz** [comment]
Yes, ok, I see the issue.
I researched it a bit more, and found that those hooks are a classic solution.
However, I also found that there already are a few variables in the build environment: https://docs.docker.com/docker-hub/builds/advanced/
Did you already know that?

The `SOURCE_COMMIT` and `IMAGE_NAME` apparently already are supplied. You only miss the BUILD_DATE. However, I'm not sure we reeeeeallly need to put it in, as the commit is enough to find the date if necessary.

What do you say we give it a try with those build environment variables?

File: Dockerfile (hunk 1 above)
Comment (silverdaz) [reply Dockerfile]: Do use `3.6.16` and not `3.6`.
The reason is in PR#4's comments.

**Comment by silverdaz** [comment]
And by the way, why not fetch the BUILD_DATE from the Dockerfile. Don't you have the `date` command in the build environment? If not, then add it, use it and toss it. You'll have the BUILD_DATE properly setup. No need for the hook.

Am I missing something?

**Comment by jbygdell** [comment]
> Yes, ok, I see the issue.
> I researched it a bit more, and found that those hooks are a classic solution.
> However, I also found that there already are a few variables in the build environment: https://docs.docker.com/docker-hub/builds/advanced/
> Did you already know that?
Yes.
> 
> The `SOURCE_COMMIT` and `IMAGE_NAME` apparently already are supplied. You only miss the BUILD_DATE. However, I'm not sure we reeeeeallly need to put it in, as the commit is enough to find the date if necessary.

`SOURCE_COMMIT` is in it's default setting the full length sha, this is not needed, the short version is enough, hence the redefinition in the hook.

`IMAGE_NAME` needs to be passed when building via hooks to get the proper name of the image, this is done by Docker Hub normally but since we override the default build command we need to pass this variable to the build command. If you look you will see that this is never defined, only passed from the environment into the build command.

`BUILD_DATE` needs to be passed as an ARG to the build environment as an regular variable export inside the Docker file won't persist across layer boundaries.

**Comment by silverdaz** [comment]
Ok, good.

Do address the version number, and we're good to go.

**Pull request merged**